addons/binding/org.openhab.binding.cups/src/main/java/org/openhab/binding/cups/handler/CupsBridgeHandler.java

```diff
@@ -0,0 +1,171 @@
+	public int refreshInterval;
+
+	private CupsClient client;
+
+	private HashSet<String> lastActiveDevices = new HashSet<String>();
+
+	private List<PrinterStatusListener> deviceStatusListeners = new CopyOnWriteArrayList<>();
+
+	private ScheduledFuture<?> pollingJob;
+	private Runnable pollingRunnable = new Runnable() {
+		@Override
+		public void run() {
+			doRefresh();
+		}
+	};
+
+	public CupsBridgeHandler(Bridge bridge) {
+		super(bridge);
+	}
+	
+	protected void doRefresh() {
+		try {
+			for (CupsPrinter device : client.getPrinters()) {
+				if (lastActiveDevices != null
+						&& lastActiveDevices.contains(device.getName())) {
+					for (PrinterStatusListener deviceStatusListener : deviceStatusListeners) {
+						try {
+							deviceStatusListener.onDeviceStateChanged(
+									getThing().getUID(), device);
+						} catch (Exception e) {
+							logger.error(
+									"An exception occurred while calling the DeviceStatusListener",
+									e);
+						}
+					}
+				} else {
+					for (PrinterStatusListener deviceStatusListener : deviceStatusListeners) {
+						try {
+							deviceStatusListener.onDeviceAdded(getThing(),
+									device);
+							deviceStatusListener.onDeviceStateChanged(
+									getThing().getUID(), device);
+						} catch (Exception e) {
+							logger.error(
+									"An exception occurred while calling the DeviceStatusListener",
+									e);
+						}
+						lastActiveDevices.add(device.getName());
+					}
+				}
+			}
+		} catch (Exception e) {
+			logger.error("exception while getting printer from {}: {}",
+					client, e.getMessage());
+		}
+	}
+	
+	public CupsClient getClient() {
+		return client;
+	}
+
+	@Override
+	public void handleCommand(ChannelUID channelUID, Command command) {
+		if (command instanceof RefreshType) {
+			logger.debug("Refresh command received.");
+			doRefresh();
+		} else
+			logger.warn("No bridge commands defined.");
+	}
+
+	private synchronized void startAutomaticRefresh() {
+		if (pollingJob == null || pollingJob.isCancelled()) {
+			pollingJob = scheduler.scheduleAtFixedRate(pollingRunnable, 0,
+					refreshInterval, TimeUnit.MILLISECONDS);
+		}
+	}
+
+	@Override
+	public void initialize() {
+		logger.debug("Initializing Cups bridge handler.");
+		Configuration conf = this.getConfig();
+		super.initialize();
+		this.host = String.valueOf(conf.get(CupsBindingConstants.BRIDGE_PARAMETER_HOST));
+		try {
+			this.port = Integer.parseInt(String.valueOf(conf
+					.get(CupsBindingConstants.BRIDGE_PARAMETER_PORT)));
+		} catch (Exception ex) {
+			this.port = 631;
+		}
+		try {
+			this.refreshInterval = Integer.parseInt(String.valueOf(conf
+					.get(CupsBindingConstants.BRIDGE_PARAMETER_REFRESH_INTERVAL)));
+		} catch (Exception ex) {
+			this.refreshInterval = 30000;
+		}
+		if (host != null && host != "") {
+			try {
+				client = new CupsClient(host, port);
+				startAutomaticRefresh();
+			} catch (Exception e) {
+				logger.error("Couldn't connect to Cups server [Host '"
```

Comment:
It cannot connect at the moment. But maybe the server is down just now and back in a minute? So probably you should also schedule a job that regularly retries if it can establish a server communication.